Neo4j.Driver/Neo4j.Driver/Internal/Routing/ClusterDiscoveryManager.cs

```diff
@@ -52,35 +57,10 @@ public ClusterDiscoveryManager(IConnection connection, IDictionary<string, strin
 
         /// <remarks>Throws <see cref="ProtocolException"/> if the discovery result is invalid.</remarks>
         /// <remarks>Throws <see cref="ServiceUnavailableException"/> if the no discovery procedure could be found in the server.</remarks>
-        public void Rediscovery()
-        {
-            try
-            {
-                using (var provider = new SingleConnectionBasedConnectionProvider(_conn))
-                using (var session = new Session(provider, _logger))
-                {
-                    var result = session.Run(DiscoveryProcedure);
-                    var record = result.Single();
-                    ParseDiscoveryResult(record);
-                }
-            }
-            catch (Exception e)
-            {
-                HandleDiscoveryException(e);
-            }
-
-            if (!Readers.Any() || !Routers.Any())
-            {
-                throw new ProtocolException(
-                    $"Invalid discovery result: discovered {Routers.Count()} routers, " +
-                    $"{Writers.Count()} writers and {Readers.Count()} readers.");
-            }
-        }
-
         public async Task RediscoveryAsync()
         {
             var provider = new SingleConnectionBasedConnectionProvider(_conn);
-            var session = new Session(provider, _logger);
+            var session = new Session(provider, _logger, _syncExecutor);
```

Comment:
Seems this is not necessary needed as we are only using the async part. Maybe a dummy executor which will throw exception if we try to do sync operation here is enough. I just feel a lot of effort to pass this `_syncExecutor` down but we actually do not need it.